Hide the USD component's Util.hh file from public API

adlarkin opened the pull request:
Signed-off-by: Ashton Larkin <42042756+adlarkin@users.noreply.github.com>

# 🦟 Bug fix

## Summary
When I originally worked on #818, I created a set of helper methods for achieving generic parsing tasks between SDF and USD. These methods were placed in `usd/include/sdf/usd/sdf_parser/Utils.hh`. After #818 was merged, I realized that the `Utils.hh` is meant for internal use only, and should not be exposed to downstream users. So, this PR moves `Utils.hh` to `usd/src`, and also updates the methods in this file to make use of the `UsdError` class that was introduced in #836.

_It should be noted that the ABI checker may fail since I am removing methods from the public API. However, I talked with others offline, and they said that for a scenario like this, it's okay because the public API I am changing in the USD component has not been released yet. So, we will have to ignore the ABI checker for this PR._

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

scpeters reviewed (commented):
I wouldn't expect the ABI checker to fail since that CI job does not have the `pxr` dependency installed or built from source

also, it looks like this isn't simply a move of the header file; it looks like you made some changes to some of the methods in Utils.hh as well. This could be fine, but you should explain them in the pull request description

adlarkin commented:
> I wouldn't expect the ABI checker to fail since that CI job does not have the `pxr` dependency installed or built from source

You're right, ABI checker didn't fail :+1: 

> also, it looks like this isn't simply a move of the header file; it looks like you made some changes to some of the methods in Utils.hh as well. This could be fine, but you should explain them in the pull request description

Yes, you are right, there are some slight changes. They're in the PR description if you look at the end of the first paragraph:
```
also updates the methods in this file to make use of the UsdError class that was introduced in https://github.com/ignitionrobotics/sdformat/pull/836.
```

Let me know if you think any other descriptions/documentation needs to be added.

adlarkin commented:
@scpeters @ahcorde I renamed the file from `Utils.hh` to `UsdUtils.hh` and added tests for the methods in this file in https://github.com/ignitionrobotics/sdformat/pull/850/commits/33beba4e38809343423965fa7d26a2381b8f5c3a. This should be ready for review if you wouldn't mind taking a look.

ahcorde commented:
@osrf-jenkins retest this please

scpeters commented on usd/src/UsdUtils.hh:
```diff
@@ -45,38 +47,62 @@ namespace sdf
   namespace usd
   {
-    /// \brief Get an object's pose w.r.t. its parent.
+    /// \brief Set an object's pose w.r.t. its parent.
```
is this really `Set`? It seems more like it's getting the pose, since `_pose` is an output of this function

osrf-triage commented:
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1
